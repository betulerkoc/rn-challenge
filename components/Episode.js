import React from 'react';
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';

const Episode = ({title, id, navigation}) => {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={styles.button}
        onPress={() =>
          navigation.navigate('Episode Detail', {
            id: id,
          })
        }>
        <Text>{title}</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 10,
  },
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 10,
    borderWidth: 2,
    borderColor: '#0099cc',
    backgroundColor: '#e6ffff',
    borderRadius: 10,
    height: 80
  },
});

export default Episode;
