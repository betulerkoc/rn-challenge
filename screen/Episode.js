import React, {useState, useEffect} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  View,
  FlatList,
  Image,
  Dimensions,
} from 'react-native';

import Button from '../components/Button';

const WINDOW = Dimensions.get('window');

const Episode = ({navigation}) => {
  const [episode, setEpisodes] = useState(null);

  const fetchData = async () => {
    const response = await fetch('https://rickandmortyapi.com/api/episode');
    const json = await response.json();
    setEpisodes({data: json.results});
  };

  useEffect(() => {
    fetchData();
  }, []);

  return (
    <SafeAreaView style={styles.view}>
      <Image
        style={styles.logo}
        source={require('../image/rick-and-morty.png')}
      />
      {episode && (
        <View style={styles.container}>
          <FlatList
            data={episode.data}
            keyExtractor={(x, i) => i}
            renderItem={({item}) => (
              <Button title={item.name} id={item.id} navigation={navigation} />
            )}
          />
        </View>
      )}
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  view: {
    backgroundColor: '#66ff66',
  },
  container: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  logo: {
    width: WINDOW.width,
    height: 100,
  },
});

export default Episode;
