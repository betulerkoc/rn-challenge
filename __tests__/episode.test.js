import 'react-native';
import React from 'react';
import Episode from '../components/Episode';
import renderer from 'react-test-renderer';

test('Episode snapshot', () => {
    const snap = renderer.create(
        <Episode/>
    ).toJSON();
    expect(snap).toMatchSnapshot();
}) 