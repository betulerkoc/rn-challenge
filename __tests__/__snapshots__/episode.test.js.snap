// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 10,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e6ffff",
        "borderColor": "#0099cc",
        "borderRadius": 10,
        "borderWidth": 2,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
</View>
`;
